config/v1/types_network.go

```diff
@@ -13,20 +13,50 @@ type Network struct {
 	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// spec holds user settable values for configuration
+	// Spec holds user settable values for configuration.
 	Spec NetworkSpec `json:"spec"`
-	// status holds observed values from the cluster. They may not be overridden.
+	// Status holds observed values from the cluster. They may not be overridden.
 	Status NetworkStatus `json:"status"`
 }
 
+// NetworkSpec is the desired network configuration.
+// As a general rule, this SHOULD NOT be used directly. Instead, you should
+// consume the NetworkStatus, as it indicates the currently deployed configuration.
 type NetworkSpec struct {
-	// serviceCIDR
-	// servicePortRange
-	// vxlanPort
-	// 	ClusterNetworks    []ClusterNetworkEntry `json:"clusterNetworks"`
+	// IP address pool to use for pod IPs.
+	ClusterNetworks []ClusterNetwork `json:"clusterNetworks"`
+
+	// IP address pool for services.
+	// Currently, we only support a single entry here.
+	ServiceNetworks []string `json:"serviceNetworks"`
+
+	// NetworkType is the plugin that is to be deployed (e.g. OpenShiftSDN).
+	// N.B. changing this is currently not supported.
+	NetworkType string `json:"networkType"`
 }
 
+// NetworkStatus is the current network configuration.
 type NetworkStatus struct {
+	// IP address pool to use for pod IPs.
+	ClusterNetworks []ClusterNetwork `json:"clusterNetworks"`
+
+	// IP address pool for services.
+	// Currently, we only support a single entry here.
+	ServiceNetworks []string `json:"serviceNetworks"`
+
+	// NetworkType is the plugin that is deployed (e.g. OpenShiftSDN).
+	NetworkType string `json:"networkType"`
+}
+
+// ClusterNetwork is a contiguous block of IP addresses from which pod IPs are
+// allocated.
+type ClusterNetwork struct {
+	// The complete block for pod IPs.
+	CIDR string `json:"cidr"`
+
+	// The size of subnet (in bits) to allocate to each node for pod allocations.
+	// For example, 8 indicates a /24 will be given to each node.
+	HostSubnetLength uint32 `json:"hostSubnetLength"`
```

Comment:
I have always hated `HostSubnetLength`'s gratuitous backwardness, and managed to fix it when we added multiple CIDR support to ovn-kubernetes, where you now actually say "`24`" when you want a /24, rather than saying "`8`" to mean you want a /24. We should do that here too... and then we could perhaps even steal the "double CIDR" syntax that we use in the ovn-kubernetes config, and just specify each cluster network subset as a string, eg, `"10.128.0.0/14/23"` rather than needing a struct type? (If we keep the struct, then we need to rename `HostSubnetLength` too, to make it clear that it's not representing the same value as network.openshift.io.ClusterNetwork.HostSubnetLength. Maybe `HostSubnetPrefix`. Or `NodePrefix` for that matter.)